### PR TITLE
Fix compile errors with gcc8 and fix the sensors on Asus T100HA laptops not working

### PR DESCRIPTION
**Pull request by jwrdegoede**
Hi Bastien,

I've been working today on getting the sensors on Asus T100HA devices to work properly, the second commit in this set is the result of this.

When I went to test my patch I would not compile, so the first patch in this set fixes compiling with gcc8.

Regards,

Hans

### CODE REVIEW

**Comment by jwrdegoede** [comment]
Hmm, I'm testing some sensor stuff on another device now (Lenovo Miix 320), the sensor there only does polling, but it does have a sampling_frequency attribute (without a prefix) and that one defaults to 1 (Hz) which is somewhat slow. I wonder if we should do something about that too?

File: src/iio-buffer-utils.c [hunk 1]
```diff
@@ -568,6 +568,51 @@ process_scan_1 (char              *data,
...
+	device_dir = g_udev_device_get_sysfs_path (dev);
+	dir = g_dir_open (g_udev_device_get_sysfs_path (dev), 0, &error);
+	if (!dir) {
+		g_warning ("Failed to open directory '%s': %s", device_dir, error->message);
+		g_error_free (error);
+		return FALSE;
+	}
+
+	while ((name = g_dir_read_name (dir))) {
+		if (g_str_has_suffix (name, "sampling_frequency") == FALSE)
```
Comment (hadess) [inline src/iio-buffer-utils.c]: What would be the names of the files? Are we going to have multiple files with that suffix?

Comment (jwrdegoede) [reply src/iio-buffer-utils.c]: The Lenovo Miix 320 with the SMO8840 accelerometer which uses st_accel_i2c.c has just a  single "sampling_frequency" attribute.

The HID sensors on the Asus T200HA have an in_accel_sampling_frequency attribute for the accelerometer part of the sensor collection and in_magn_sampling_frequency and in_rot_sampling_frequency attributes for the compass part of the sensor collection.

**Comment by jwrdegoede** [comment]
Ok, I just pushed a new version adding the exact compile error from gcc (and dropping the .) and adding a #define for the 10 Hz minimum sample-freq.

**Comment by hadess** [comment]
> one which actually expects arguments, causing it to throw erros like these:

typo, "errors"

File: src/iio-buffer-utils.c [hunk 2]
```diff
@@ -17,6 +17,8 @@
 #include <errno.h>
 #include <stdio.h>
 
+#define IIO_MIN_SAMPLING_FREQUENCY	10
```
Comment (hadess) [inline src/iio-buffer-utils.c]: Can you please add a comment about the unit (eg. `/* Hz */`)

**Comment by hadess** [comment]
Looks good to merge after those.

**Comment by jwrdegoede** [comment]
On 17-04-18 12:41, hadess wrote:
> typo, "errors"

Fixed.

**Comment by jwrdegoede** [comment]
I was too quick :) Also added the /* Hz */ comment now.

**Comment by hadess** [comment]
As usual, thanks! :)

**Pull request merged**